packages/@aws-cdk/core/lib/annotations.ts

```diff
@@ -49,6 +51,38 @@ export class Annotations {
     this.addMessage(cxschema.ArtifactMetadataEntryType.ERROR, message);
   }
 
+  /**
+   * Adds a deprecation warning for a specific API.
+   *
+   * Deprecations will be added only once per construct as a warning and will be
+   * deduplicated based on the `api`.
+   *
+   * If the environment variable `CDK_BLOCK_DEPRECATIONS` is set, this method
+   * will throw an error instead with the deprecation message.
+   *
+   * @param api The API being deprecated in the format `module.Class.property`
+   * (e.g. `@aws-cdk/core.Construct.node`).
```

Comment:
Define a `DeprecationOptions` and add make these separate parameters.

Reply:
Why? I think it's rather intuitive that these are positional.

1. What's deprecated?
2. What's the alternative?

Both are "intrinsic" and both are required. Usually an `options` strict is useful when some inputs are _optional_.